File: helloworld.py
from threading import Thread

i = 0

def tellOpp():
	global i
	for j in range(0, 1000000):
		i += 1
	
def tellNed():
	global i
	for j in range(0, 1000000):
		i -= 1

def main():
	ink = Thread(target = tellOpp, args = (),)
	dek = Thread(target = tellNed, args = (),)
	ink.start()
	dek.start()
	ink.join()
	dek.join()
	print(i)

main()

File: helloworld.c
// gcc 4.7.2 +
// gcc -std=gnu99 -Wall -g -o helloworld_c helloworld.c -lpthread

#include <pthread.h>
#include <stdio.h>

int i=0;

void* tellOpp(){
	for (int j=0; j<1000000; j++)
		i++;
	return NULL
}

void* tellNed(){
	for (int j=0; j<1000000; j++)
		i--;
	return NULL
}

int main(){
	pthread_t ink;
	pthread_t dek;

	pthread_create(&ink, NULL, tellOpp, NULL);
	pthread_create(&ink, NULL, tellOpp, NULL);

	pthread_join(ink, NULL);
	pthread_join(dek, NULL);
	printf(i);

	return 0;
}
